### ISSUE TITLE
Using Redis queue throws 404

### ISSUE THREAD

**Issue by rrjanbiah** [pennersr/shove#4]
Apologies, if this any issue with the calling syntax.

**Setup**: Using FCM alone with `shove`

It works without Redis, but with Redis, it throws 404. Also, it misses `FCM workers started` message

### Works without Redis

```
shove -api-addr x.x.x.x:8322 -fcm-api-key xxxxxx
2020/12/29 15:00:53 Using non-persistent in-memory queue
2020/12/29 15:00:53 Initializing FCM service
2020/12/29 15:00:53 Shove server started
2020/12/29 15:00:53 FCM workers started
```

```
curl  -i  --data '{"to": "XXXXX", "notification": {"title": "Hello"}}' http://x.x.x.x:8322/api/push/fcm

HTTP/1.1 202 Accepted
Date: Tue, 29 Dec 2020 14:47:44 GMT
Content-Length: 2
Content-Type: text/plain; charset=utf-8

OK
```

### Not working with Redis

```
shove -api-addr x.x.x.x:8322 -queue-redis redis://redis:6379 -fcm-api-key xxxxx
2020/12/29 15:04:46 Using Redis queue at redis://redis:6379
2020/12/29 15:04:46 Initializing FCM service
2020/12/29 15:04:46 Shove server started
```
```
curl  -i  --data '{"to": "XXXXX", "notification": {"title": "Hello"}}' http://x.x.x.x:8322/api/push/fcm


HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Tue, 29 Dec 2020 15:10:09 GMT
Content-Length: 19

404 page not found
```

#### Other attempts

Instead of `redis://redis:6379`, tried `redis://localhost:6379` and `localhost:6379` too.

**Comment by pennersr** [comment]
I can confirm that this happens if Redis is not reachable -- it hangs at `queueFactory.NewQueue()` because Redis is not responding. 

If you run Redis like this on your host:

```docker run --rm -p 6379:6379 redis```

Then...

```./shove -api-addr 127.0.0.1:8322 -fcm-api-key xxxxxx -queue-redis redis://localhost:6379```

... works as expected.

**Comment by pennersr** [comment]
Fixed via 0467943772088c6817b83aa2e84d0dd7b7cd7bc5
